clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -307,7 +307,12 @@ void SYCL::fpga::BackendCompiler::ConstructJob(Compilation &C,
     std::string Filename(II.getFilename());
     if (II.getType() == types::TY_Tempfilelist)
       ForeachInputs.push_back(II);
-    CmdArgs.push_back(C.getArgs().MakeArgString(Filename));
+    // Add any FPGA library lists.  These come in as special tempfile lists.
+    if (II.getType() == types::TY_TempAOCOfilelist)
```

Comment:
```suggestion
    else if (II.getType() == types::TY_TempAOCOfilelist)
```

Reply:
Done.

Reply:
Had to revert this change.  This was preventing the file from being added and broke a test.  @Fznamznon, Is it expected to have the listfile added to the foreach as well as the aoc input?

Reply:
I was totally incorrect here, sorry for that. Our discussion around line 3880 was relevant, though.